plugin/metadata/request.go

```diff
@@ -0,0 +1,212 @@
+package metadata
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/coredns/coredns/request"
+	"github.com/miekg/dns"
+)
+
+const (
+	typeEDNS0Bytes = iota
+	typeEDNS0Hex
+	typeEDNS0IP
+)
+
+const (
```

Comment:
These define the format of data in EDNS0.
Right now should be the same as REWRITE that do (for now) the same operation.

Once in Metadata, it would be expected that every plugins rather use the Metadata instead or redefining a way to extract the data from EDNS0 records.